apps/language_server/lib/language_server/providers/hover.ex

```diff
@@ -44,14 +46,121 @@ defmodule ElixirLS.LanguageServer.Providers.Hover do
     end)
   end
 
-  defp contents(%{docs: "No documentation available\n"}) do
+  defp contents(%{docs: "No documentation available\n"}, _subject \\ "") do
     []
   end
 
-  defp contents(%{docs: markdown}) do
+  defp contents(%{docs: markdown}, subject) do
     %{
       kind: "markdown",
-      value: markdown
+      value: add_hexdocs_link(markdown, subject)
     }
   end
+
+  defp add_hexdocs_link(markdown, subject) do
+    [hd | tail] = markdown |> String.split("\n\n")
+
+    link = hexdocs_link(markdown, subject)
+
+    case link do
+      "" ->
+        markdown
+
+      _ ->
+        hd <> "  [view on hexdocs](#{link})\n\n" <> Enum.join(tail, "")
+    end
+  end
+
+  defp hexdocs_link(markdown, subject) do
+    t = markdown |> String.split("\n\n") |> hd() |> String.replace(">", "") |> String.trim()
+    dep = subject |> root_module_name() |> dep_name()
+
+    cond do
+      func?(t) ->
+        if dep != "" do
+          "#{@hex_base_url}/#{dep}/#{module_name(subject)}.html##{func_name(subject)}/#{params_cnt(t)}"
+        else
+          ""
+        end
+
+      true ->
+        if dep != "" do
+          "#{@hex_base_url}/#{dep}/#{t}.html"
+        else
+          ""
+        end
+    end
+  end
+
+  defp remove_special_symbol(s) do
+    s |> String.replace("!", "") |> String.replace("?", "")
+  end
+
+  defp func?(s) do
+    s =~ ~r/.*\..*\(.*\)/
+  end
+
+  defp module_name(s) do
+    [_ | tail] = s |> String.split(".") |> Enum.reverse()
+    tail |> Enum.reverse() |> Enum.join(".")
+  end
+
+  defp func_name(s) do
+    s |> String.split(".") |> Enum.reverse() |> hd()
+  end
+
+  defp params_cnt(s) do
+    cond do
+      true == (s =~ ~r/\(\)/) -> 0
+      false == String.contains?(s, ",") -> 1
+      true -> s |> String.split(",") |> length()
+    end
+  end
+
+  defp dep_name(root_mod_name) do
+    s = root_mod_name |> source()
+
+    cond do
+      third_dep?(s) -> third_dep_name(s)
+      builtin?(s) -> builtin_dep_name(s)
+      true -> ""
+    end
+  end
+
+  defp root_module_name(subject) do
+    subject |> String.split(".") |> hd()
+  end
+
+  defp source(mod_name) do
+    dep = ("Elixir." <> mod_name) |> String.to_atom()
+    dep.__info__(:compile) |> Keyword.get(:source) |> List.to_string()
+  end
+
+  defp third_dep?(source) do
+    prefix = File.cwd!() <> "/deps"
+    String.starts_with?(source, prefix)
+  end
+
+  defp third_dep_name(source) do
+    prefix = File.cwd!() <> "/deps/"
```

Comment:
Cannot use cwd here as it changes during compilation. Use root_path from server state instead